packages/toolpad-app/src/toolpad/propertyControls/SelectOptions.tsx

```diff
@@ -47,17 +48,38 @@ function SelectOptionsPropEditor({
     return null;
   }, [editingIndex, value]);
 
+  const watchInputValue = React.useCallback(
```

Comment:
I would call this function `validateOptionValue` or something like that, and if it found any duplicates it would set the error message to "Must not have duplicate values", otherwise an empty string ('').